synop2bufr/__init__.py

```diff
@@ -1126,55 +1116,61 @@ def transform(data: str, metadata: str, year: int,
             raise e
         # get TSI
         tsi = msg['station_id']
+
         # set WSI
         wsi = tsi_mapping[tsi]
+
         # parse WSI to get sections
         try:
             wsi_series, wsi_issuer, wsi_issue_number, wsi_local = wsi.split("-")   # noqa
+
+            # get other required metadata
+            latitude = metadata_dict[wsi]["latitude"]
+            longitude = metadata_dict[wsi]["longitude"]
+            station_height = metadata_dict[wsi]["elevation"]
+
+            # add these values to the data dictionary
+            msg['_wsi_series'] = wsi_series
+            msg['_wsi_issuer'] = wsi_issuer
+            msg['_wsi_issue_number'] = wsi_issue_number
+            msg['_wsi_local'] = wsi_local
+            msg['_latitude'] = latitude
+            msg['_longitude'] = longitude
+            conversion_success[tsi] = True
         except Exception:
-            raise ValueError(f"""Invalid WSI ({wsi}) found in station file,
-                             unable to parse""")
-
-        # get other required metadata
-        latitude = metadata_dict[wsi]["latitude"]
-        longitude = metadata_dict[wsi]["longitude"]
-        station_height = metadata_dict[wsi]["elevation"]
-
-        # add these values to the data dictionary
-        msg['_wsi_series'] = wsi_series
-        msg['_wsi_issuer'] = wsi_issuer
-        msg['_wsi_issue_number'] = wsi_issue_number
-        msg['_wsi_local'] = wsi_local
-        msg['_latitude'] = latitude
-        msg['_longitude'] = longitude
-
-        # update mappings for number of clouds
-        s3_mappings = {}
+            conversion_success[tsi] = False
+
+            if wsi == "":
+                LOGGER.error(f"Missing WSI for station {tsi}")
+            else:
+                LOGGER.error((f"Invalid WSI ({wsi}) found in station file,"
+                              " unable to parse"))
+
```

Comment:
what happens after the error is logged? do we continue trying to process the message?

Reply:
No, because `conversion_success[tsi]` is set to False, which prevents the message from being processed in the later stages.

Reply:
Ah, I see it now. Can you add all the error messages to the `['_meta']['status']['errors']` list.